packages/core/src/odata/common/payload-value-converter.ts

```diff
@@ -27,7 +27,7 @@ const fromBigNumber = (value: BigNumber): string =>
 /**
  * @hidden
  */
-export function parseNumber(value: string | number): number {
+export function toNumberFromEdm(value: string | number): number {
```

Comment:
I am not a fan of the names here, because somehow it does not resonate in my head. Alternatives:
`edmToNumber` + `numberToEdm`, `fromEdmToNumber` + `fromNumberToEdm`